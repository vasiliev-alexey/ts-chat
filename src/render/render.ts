export function render(state: State): void {
  const messagesBlock = document.querySelector(".container");
  let renderString = "";

  state.messages.forEach((el) => {
    if (el.name === state.userName) {
      renderString += `<div class='userMessage'><p style='font-weight:bold'>${el.name}</p><span>${el.message}</span></div>\n`;
    } else {
      renderString += `<div class='messageIncoming'><p style='font-weight:bold'>${el.name}</p><span>${el.message}</span></span></div>\n`;
    }
  });
  messagesBlock.innerHTML = renderString;
}
