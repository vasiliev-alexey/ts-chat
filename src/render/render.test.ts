import { render } from "./render";

describe("render function test", () => {
  it("first render test", () => {
    document.body.innerHTML = `<div class='container'></div>`;

    const messages: Message[] = [
      {
        name: "Manchini",
        message: "Win Win Win!",
      },
      {
        name: "Belotti",
        message: "Forza Italia!",
      },
    ];

    const state = { userName: "Buffon", messages };
    const result = `<div class="messageIncoming"><p style="font-weight:bold">${messages[0].name}</p><span>${messages[0].message}</span></div>\n<div class="messageIncoming"><p style="font-weight:bold">${messages[1].name}</p><span>${messages[1].message}</span></div>\n`;
    render(state);
    expect(document.querySelector(".container").innerHTML).toEqual(result);
  });
});
