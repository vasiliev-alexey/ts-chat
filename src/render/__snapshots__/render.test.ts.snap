// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render function test first render test 1`] = `
<div
  class="container"
>
  <div
    class="messageIncoming"
  >
    <p
      style="font-weight:bold"
    >
      Manchini
    </p>
    <span>
      Win Win Win!
    </span>
  </div>
  

  <div
    class="messageIncoming"
  >
    <p
      style="font-weight:bold"
    >
      Belotti
    </p>
    <span>
      Forza Italia!
    </span>
  </div>
  

</div>
`;
